### DISCUSSION TITLE
How to globally set Attributes for ALL Metrics

### DISCUSSION THREAD

**Discussion by lokeshmavale** [open-telemetry/opentelemetry-java-instrumentation#7695] [Q&A] [answered]
Hi Team,

I am using Prometheus and OTLP exporter. By using following technique, I can customize metrics and add attributes only when its otlp exporter. How do I add a hook to Prometheus exporter while using Java agent?

autoConfiguration.addMetricExporterCustomizer(this::configureSdkMetricExporterProvider);

**Comment by mateuszrzeszutek** [comment, marked as answer]
Hey @lokeshmavale ,

Please take a look at my answer in https://github.com/open-telemetry/opentelemetry-java/discussions/5163

**Reply by lokeshmavale** [reply]
Thanks you very much :)
